src/predict/jobs/forecast/daily_forecasts.py

```diff
@@ -0,0 +1,128 @@
+    print('preprocess_forecast_data completed')
+    return forecast_data
+
+
+def get_new_row(df_tmp, device, model):
+    last_row = df_tmp[df_tmp["device_number"] == device].iloc[-1]
+    new_row = pd.Series(index=last_row.index, dtype='float64')
+    new_row["created_at"] = last_row["created_at"] + pd.Timedelta(days=1)
+    new_row["device_number"] = device
+    new_row[f'pm2_5_last_1_day'] = last_row["pm2_5"]
+    new_row[f'pm2_5_last_2_day'] = last_row[f'pm2_5_last_{1}_day']
+    shifts = [3, 7, 14, 30]
+    functions = ['mean', 'std', 'max', 'min']
+    for s in shifts:
+        for f in functions:
+            if f == 'mean':
+                new_row[f'pm2_5_{f}_{s}_day'] = (last_row["pm2_5"] + last_row[f'pm2_5_{f}_{s}_day'] * (s - 1)) / s
+            elif f == 'std':
+                new_row[f'pm2_5_{f}_{s}_day'] = np.sqrt((last_row["pm2_5"] - last_row[f'pm2_5_mean_{s}_day']) ** 2 + (
+                        last_row[f'pm2_5_{f}_{s}_day'] ** 2 * (s - 1))) / s
+            elif f == 'max':
+                new_row[f'pm2_5_{f}_{s}_day'] = max(last_row["pm2_5"], last_row[f'pm2_5_{f}_{s}_day'])
+            elif f == 'min':
+                new_row[f'pm2_5_{f}_{s}_day'] = min(last_row["pm2_5"], last_row[f'pm2_5_{f}_{s}_day'])
+
+                # Use the date of the new row to create other features
+    attributes = ['year', 'month', 'day', 'dayofweek']
+    for a in attributes:
+        new_row[a] = new_row['created_at'].__getattribute__(a)
+        new_row['week'] = new_row["created_at"].isocalendar().week
+
+    new_row["pm2_5"] = model.predict(new_row.drop(["created_at", "pm2_5"]).values.reshape(1, -1))[0]
+    return new_row
+
+
+def save_next_1_week_forecast_results(data):
+        db.daily_forecasts.insert_many(data)
+        print('saved')
+
+
+def get_next_1week_forecasts(target_column, model):
+    print('Getting next 1 week forecasts')
+    forecast_data = preprocess_forecast_data(target_column)
+    test_forecast_data = forecast_data.copy()
+    forecast_horizon = 7  # number of days to forecast
+    next_1week_forecasts = pd.DataFrame()
+
+    for device in test_forecast_data["device_number"].unique():
+        test_copy = test_forecast_data[test_forecast_data["device_number"] == device]
+        for i in range(forecast_horizon):
+            new_row = get_new_row(test_copy, device, model)
+            test_copy = pd.concat([test_copy, new_row.to_frame().T], ignore_index=True)
+        next_1week_forecasts = pd.concat([next_1week_forecasts, test_copy], ignore_index=True)
+
+    next_1week_forecasts['device_number'] = next_1week_forecasts['device_number'].astype(int)
+    next_1week_forecasts['pm2_5'] = next_1week_forecasts['pm2_5'].astype(float)
+    return next_1week_forecasts[['created_at', 'pm2_5', 'device_number']][
+        next_1week_forecasts['created_at'] > configuration.TEST_DATE_DAILY_START]
+
+
+if __name__ == '__main__':
+    TARGET_COL = 'pm2_5'
+    # TODO: Remove for deployment use GCS
```

Comment:
Read the model from the Bucket 
model = get_trained_model_from_gcs(configuration.GOOGLE_CLOUD_PROJECT_ID, configuration.AIRQO_PREDICT_BUCKET,
                                     'daily_forecast_model.pkl')